doc: Add DeepSeek-R1 perf doc

Kefeng-Duan opened the pull request.

FrankD412 commented on docs/source/blogs/Best_perf_practice_on_DSR1_in_TRT-LLM.md:
```diff
@@ -0,0 +1,122 @@
+
+DS_R1_NVFP4_ALLMOE_MODEL_PATH=$YOUR_MODEL_PATH/DeepSeek-R1-FP4
+trtllm-bench --model deepseek-ai/DeepSeek-R1 \
+    --model_path $DS_R1_NVFP4_ALLMOE_MODEL_PATH \
```
If we're running FP4, we can just use the Huggingface name here directly, `nvidia/DeepSeek-R1-FP4`.

For FP8, are we going to be expecting a user to specify an `hf_quant_config.json` in their Huggingface checkpoint directory? Right now the only way to activate FP8 is with an Nvidia quantized checkpoint.

FrankD412 commented on docs/source/blogs/Best_perf_practice_on_DSR1_in_TRT-LLM.md:
````diff
@@ -0,0 +1,122 @@
+```
+
+Explanation:
+- `trtllm-bench`: A CLI packags benchmarking utility that aims to make it easier for users to reproduce our officially published. [TensorRT-LLM Benchmarking](https://nvidia.github.io/TensorRT-LLM/performance/perf-benchmarking.html).
````
```suggestion
- `trtllm-bench`: A CLI benchmarking utility that aims to make it easier for users to reproduce our officially published. [TensorRT-LLM Benchmarking](https://nvidia.github.io/TensorRT-LLM/performance/perf-benchmarking.html).
```

FrankD412 commented on docs/source/blogs/Best_perf_practice_on_DSR1_in_TRT-LLM.md:
```diff
@@ -0,0 +1,122 @@
+Explanation:
+- `trtllm-bench`: A CLI packags benchmarking utility that aims to make it easier for users to reproduce our officially published. [TensorRT-LLM Benchmarking](https://nvidia.github.io/TensorRT-LLM/performance/perf-benchmarking.html).
+- `--dataset`: Prompt dataset used to benchmark. our official benchmark dataset has ISL = 1K, OSL = 2K
+- `--backend`: Inference backend. Here we use Pytorch backed. 
```
```suggestion
- `--backend`: Inference backend. Here we use Pytorch backend. 
```

FrankD412 commented on docs/source/blogs/Best_perf_practice_on_DSR1_in_TRT-LLM.md:
```diff
@@ -0,0 +1,122 @@
+
+# Clone the DeepSeek-R1-FP4 model
+cd $YOUR_MODEL_PATH
+GIT_LFS_SKIP_SMUDGE=1 git clone https://huggingface.co/nvidia/DeepSeek-R1-FP4
```
See my comment below -- we should be able to use the model name directly.

kaiyux commented on docs/source/blogs/Best_perf_practice_on_DSR1_in_TRT-LLM.md:
```diff
@@ -0,0 +1,122 @@
+# How to get best performance on DSR1 in TRT-LLM
```
nit: this is what I learnt from previous interaction with technical writer - let's never use terms like "TRT-LLM" in the documents, but always use "TensorRT-LLM".

kaiyux commented on docs/source/blogs/Best_perf_practice_on_DSR1_in_TRT-LLM.md:
```diff
@@ -0,0 +1,122 @@
+
+NVIDIA has announced world-record DeepSeek-R1 inference performance at NVIDIA GTC 2025. A single NVIDIA DGX system with eight NVIDIA Blackwell GPUs can achieve over 250 tokens per second per user or a maximum throughput of over 30,000 tokens per second on the massive, state-of-the-art 671 billion parameter DeepSeek-R1 model. [NVIDIA Blackwell Delivers World-Record DeepSeek-R1 Inference Performanc ...
+
+In this blog, we share the configrations and procedures about how to reproduce the number on both B200 and H200 with Pytorch workflow.
```
Copilot is making a valid point here : )
> Typo: 'configrations' should be corrected to 'configurations'.

jiahanc commented on docs/source/blogs/Best_perf_practice_on_DSR1_in_TRT-LLM.md:
````diff
@@ -0,0 +1,228 @@
+YOUR_MODEL_PATH=<YOUR_MODEL_PATH>
+YOUR_DATA_PATH=<YOUR_DATA_PATH>
+
+# Clone the TensorRT-LLM repository
+cd $YOUR_WORK_PATH
+git clone https://github.com/NVIDIA/TensorRT-LLM.git
+cd TensorRT-LLM
+git submodule update --init --recursive
+git lfs pull
+
+# Clone the DeepSeek-R1-FP4 model
+cd $YOUR_MODEL_PATH
+GIT_LFS_SKIP_SMUDGE=1 git clone https://huggingface.co/nvidia/DeepSeek-R1-FP4
+git lfs pull  # Download the full model weight will take a long time
+```
+**Note**: Replace `<*_PATH>` to your actual path. 
+
+### Build Docker 
+Create a docker and run:
+
+``` bash
+cd TensorRT-LLM
+make -C docker jenkins_run LOCAL_USER=1 DOCKER_RUN_ARGS="-v $YOUR_MODEL_PATH:$YOUR_MODEL_PATH:ro -v $YOUR_DATA_PATH:$YOUR_DATA_PATH:ro -v $YOUR_WORK_PATH:$YOUR_WORK_PATH"
+```
+Here we set `LOCAL_USER=1` argument to set up the local user account inside the container.
+
+### Compile and Install
+Here we compile the source inside the container:
+
+``` bash
+python3 ./scripts/build_wheel.py --trt_root /usr/local/tensorrt --benchmarks --use_ccache --cuda_architectures "100-real"  --python_bindings --clean
+```
+
+Install and set environment variables:
+
+```bash
+pip install --user build/tensorrt_llm*.whl
+export PATH=${HOME}/.local/bin:${PATH}
+export PYTHONPATH=`pwd`
+```
````
i think this Prerequisites part can also be sued in the throughput guide, shall we move this above the latency and throughput subtitle and set as Prerequisites?

jiahanc commented on docs/source/blogs/Best_perf_practice_on_DSR1_in_TRT-LLM.md:
```diff
@@ -0,0 +1,228 @@
+# How to get best performance on DSR1 in TRT-LLM
+
+NVIDIA has announced world-record DeepSeek-R1 inference performance at NVIDIA GTC 2025. A single NVIDIA DGX system with eight NVIDIA Blackwell GPUs can achieve over 250 tokens per second per user or a maximum throughput of over 30,000 tokens per second on the massive, state-of-the-art 671 billion parameter DeepSeek-R1 model. [NVIDIA Blackwell Delivers World-Record DeepSeek-R1 Inference Performanc ...
```
since we are also documenting Hopper, shall we also mention hopper performance number here? 
If so, we may want to reorganize the structure of this paragraph：）

hlu1 commented on docs/source/blogs/Best_perf_practice_on_DSR1_in_TRT-LLM.md:
````diff
@@ -0,0 +1,362 @@
+```
+
+## H200 NVL8
+### Prerequisites
+
+``` bash
+# Prerequisites
+apt-get update && apt-get -y install git git-lfs
+git lfs install
+
+# Replace with your actual path
+YOUR_WORK_PATH=<YOUR_WORK_PATH>
+YOUR_MODEL_PATH=<YOUR_MODEL_PATH>
+YOUR_DATA_PATH=<YOUR_DATA_PATH>
+
+# Clone the TensorRT-LLM repository
+cd $YOUR_WORK_PATH
+git clone https://github.com/NVIDIA/TensorRT-LLM.git
+cd TensorRT-LLM
+git submodule update --init --recursive
+git lfs pull
+
+# Clone the DeepSeek-R1 model
+cd $YOUR_MODEL_PATH
+GIT_LFS_SKIP_SMUDGE=1 git clone https://huggingface.co/deepseek-ai/DeepSeek-R1
+git lfs pull  # Download the full model weight will take a long time
+```
+**Note**: Replace `<*_PATH>` to your actual path. 
+
+#### Build Docker 
+Create a docker and run:
+
+``` bash
+cd TensorRT-LLM
+make -C docker jenkins_run LOCAL_USER=1 DOCKER_RUN_ARGS="-v $YOUR_MODEL_PATH:$YOUR_MODEL_PATH:ro -v $YOUR_DATA_PATH:$YOUR_DATA_PATH:ro -v $YOUR_WORK_PATH:$YOUR_WORK_PATH"
+```
+Here we set `LOCAL_USER=1` argument to set up the local user account inside the container.
````
There is no need to repeat these duplicated instructions. These should be common instructions regardless of B200/H200.

hlu1 commented on docs/source/blogs/Best_perf_practice_on_DSR1_in_TRT-LLM.md:
````diff
@@ -0,0 +1,362 @@
+```
+
+Explanation:
+- `trtllm-bench`: A CLI packags benchmarking utility that aims to make it easier for users to reproduce our officially published. [TensorRT-LLM Benchmarking](https://nvidia.github.io/TensorRT-LLM/performance/perf-benchmarking.html).
+- `--dataset`: Prompt dataset used to benchmark. our official benchmark dataset has ISL = 1K, OSL = 2K
+- `--backend`: Inference backend. Here we use Pytorch backed. 
+- `--tp 8`: Tensor parallel size is 8.
+- `--ep 4`: Expert parallel size is 4.
+- `--extra_llm_api_options`: Used to specify some extra config. The content of the file is as follows:
+
+    ``` yaml
+    pytorch_backend_config:
+        enable_overlap_scheduler: true
+        use_cuda_graph: true
+    speculative_config:
+        decoding_type: MTP
+        num_nextn_predict_layers: 3
+    ```
````
These are exactly the same as B200. No need to duplicate the exact the instructions. We should refactor the instructions and only mention the difference on H200 when necessary.

Kefeng-Duan commented:
/bot skip

Kefeng-Duan commented:
/bot skip

litaotju committed:
Update README and add benchmarking blog for DeepSeek-R1

- Added a new entry in the README for the published benchmarking best practices for DeepSeek-R1.
- Introduced a new blog post detailing performance benchmarking configurations and procedures for DeepSeek-R1 in TensorRT-LLM, including installation, dataset preparation, and benchmarking steps for both B200 and H200 GPUs.

Signed-off-by: taoli <litaotju@users.noreply.github.com>

litaotju commented:
/bot run --help

litaotju commented:
/bot run

tensorrt-cicd commented:
[PR_Github #1714](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1714/) Bot args parsing error!

kaiyux commented:
/bot skip --comment "doc changes"

tensorrt-cicd commented:
[PR_Github #1716](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1716/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #1717](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1717/) [ skip ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #1716](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1716/) [ run ] completed with state `ABORTED`

tensorrt-cicd commented:
[PR_Github #1717](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1717/) [ skip ] completed with state `SUCCESS`
[Release Check Pipeline #357](https://nv/trt-llm-cicd/job/helpers/job/ReleaseCheck_PR/357/) failed

litaotju committed:
Fix format

Signed-off-by: taoli <litaotju@users.noreply.github.com>

litaotju commented:
/bot skip --comment "doc changes"

tensorrt-cicd commented:
[PR_Github #1736](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1736/) [ skip ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #1737](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1737/) [ skip ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #1736](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1736/) [ skip ] completed with state `ABORTED`

tensorrt-cicd commented:
[PR_Github #1737](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1737/) [ skip ] completed with state `ABORTED`

kaiyux commented:
/bot skip --comment "doc changes"

tensorrt-cicd commented:
[PR_Github #1739](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1739/) [ skip ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #1739](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1739/) [ skip ] completed with state `SUCCESS`
Skipping testing for commit `ef556ab`

tensorrt-cicd commented:
[PR_Github #1748](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1748/) [ skip ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #1748](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1748/) [ skip ] completed with state `SUCCESS`
Skipping testing for commit `70712ab`
